### DISCUSSION TITLE
Localisation

### DISCUSSION THREAD

**Discussion by ghost** [curl/curl#8499] [Ideas]
It will be useful if everyone in the world can use curl in their own language

**Comment by dfandrich** [comment]
I was wondering when this was going to be asked. curl is just shy of 24 years
old, and I don't recall this ever being requested before. I started looking
into what it would take to do message localization well over a decade ago, but
stopped when I realized that nobody was asking for it. Maybe it just took
hitting the 10,000,000,000 installation threshold before someone wanted it :-)

**Comment by bagder** [comment]
I have no experience with this. How do you even do localization for a library? The mix with the (separate) command line tool is also interesting as I figure you'd want them to speak the same language. Localized formatting of output numbers (thousand separators etc) is also an added complexity and risk breaking scripts across the world.

**Comment by danielgustafsson** [comment]
> On 23 Feb 2022, at 10:09, Daniel Stenberg ***@***.***> wrote:

> I have no experience with this. How do you even do localization for a library? The mix with the (separate) command line tool is also interesting as I figure you'd want them to speak the same language. Localized formatting of output numbers (thousand separators etc) is also an added complexity and risk breaking scripts across the world.

GNU gettext can handle tool + library translations with translation domains. It's - from experience - a non-trivial amount of work though.

**Comment by jornfranke** [comment]
It seems to have been done for Linux command line tools (e.g. ls, cp etc.). I have seen that some distributions offer those in other languages than English. I guess they use the same/similar tools as mentioned above.

I have noticed that they organize the translation via the translation project: https://translationproject.org/domain/coreutils.html

It is probably not only a technical effort, but also a significant coordination and reviewing one of the translations themselves. Github or similar are probably not suitable for this.

**Comment by jornfranke** [comment]
Nevertheless, I think the idea to support it is great :)

**Comment by roalz** [comment]
As a foreign language speaker (Italian) I like the idea, 
but as a tech person I would think twice and more before localizing/internationalizing a command-line tool such as curl.

Having curl --help and all informative text localized would be ok,
but please don't forget that curl is often used as backend to scripts and other software that may rely on curl output geing in English locale (i.e. number formatting, etc...).

As this may be a breaking change, I would suggest making it an opt-in feature (localized on user demand).

**Comment by ghost** [comment]
You can try using a platform called Crowdin for translations
